petab/lint.py

```diff
@@ -65,6 +93,79 @@ def condition_table_is_parameter_free(condition_df):
             return False
     return True
 
+def check_parameter_sheet(problem):
+    check_parameter_df(problem.parameter_df)
+    parameterId_is_string(problem.parameter_df)
+    parameterScale_is_valid(problem.parameter_df)
+    parameterBounds_are_numeric(problem.parameter_df)
+    parameterEstimate_is_boolean(problem.parameter_df)
+    parameterId_is_unique(problem.parameter_df)
+    check_parameterBounds(problem.parameter_df)
+    return True
+
+def parameterId_is_string(parameter_df):
+    """Check if all entries in the parameterId column of the parameter table are string and not empty"""
+
+    for parameterId in parameter_df['parameterId']:
+        if isinstance(parameterId, str):
+            if parameterId[0].isdigit():
+                raise ValueError('parameterId ' + parameterId + ' starts with integer')
+        else:
+            raise ValueError('Empty parameterId found')
+
+    return True
```

Comment:
I would expect `parameterId_is_string` to return false if not all parameterIds are string (instead of raising)

Reply:
Same for the functions below. If called `*is*`, they should return true or false, and raise only in case of error. Otherwise please call them `assert_*` or similar.

Reply:
What would you recommend?
if we only return False maybe this is not very informative...

Reply:
call it `assert_parameter_name_is_valid()`?

Reply:
(And no return value should be checked for in the caller, obviously)